presto-hive/src/main/java/com/facebook/presto/hive/HiveWriter.java

```diff
@@ -110,12 +114,27 @@ public PartitionUpdate getPartitionUpdate()
                 updateMode,
                 writePath,
                 targetPath,
-                ImmutableList.of(fileWriteInfo),
+                ImmutableList.of(new FileWriteInfo(fileWriteInfo.getWriteFileName(), fileWriteInfo.getTargetFileName(), getFileSize())),
                 rowCount,
                 inputSizeInBytes,
                 fileWriter.getWrittenBytes());
     }
 
+    private Optional<Long> getFileSize()
+    {
+        if (fileStatistics.isPresent()) {
+            // FileStatistics page layout:
+            //
+            // fileSize   rowCount
+            //  X             X
+
+            Block fileSizeBlock = fileStatistics.get().getBlock(0);
+            return Optional.of(BIGINT.getLong(fileSizeBlock, 0));
```

Comment:
Move this to the same util class. We need a central place to encode and decode states info from/to pages